.github/workflows/publish.yml

```diff
@@ -0,0 +1,55 @@
+name: Publish to PyPI [Production]
+on:
+  release:
+    types: [published]
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ github.event.repository.name }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+      #-----------------------------------------------------------------------------
+      # Make sure the version is set to the current tagged release
+      #-----------------------------------------------------------------------------
+      - name: Update pyproject.toml
+        run: poetry version ${{ github.ref_name }}
+      #----------------------------------------------
+      # Attempt push to test-pypi
+      #----------------------------------------------
+      - name: Build and publish to pypi
+        uses: JRubics/poetry-publish@v1.10
+        with:
+          pypi_token: ${{ secrets.PYPI_TEST_SECRET }}
```

Comment:
This is from the code you contributed :P.  So my recent experience with test.pypi.org shows me there's very little benefit, because the dependencies aren't there, and I don't think most of our users are installing the wheel, so I'm happy to go direct to pypi.

Reply:
Two things:
- My feedback has to do with the name of the env variable. I'm fine with pushing direct to pypi.
- The reason we do test builds for pysql with test.pypi.org isn't to make an installable version (because you're correct that our dependencies aren't available on test.pypi.org). It's to verify that our package _on its own_ is capable of successfully uploading to the python server. This is a benefit to us, even if users can't install from there.